crates/asm/subprotocols/bridge-v1/src/state/bridge.rs

```diff
@@ -0,0 +1,897 @@
+    /// - The Deposit Request Transaction (DRT) spending signature is valid
+    /// - The deposit output is properly locked to the aggregated operator key
+    /// - The deposit index is unique within the deposits table
+    ///
+    /// # Parameters
+    ///
+    /// - `tx` - The Bitcoin transaction containing the deposit
+    /// - `info` - The parsed deposit information to validate
+    ///
+    /// # Returns
+    ///
+    /// - `Ok(())` - If the deposit passes all validation checks
+    /// - `Err(DepositError)` - If validation fails for any reason
+    ///
+    /// # Errors
+    ///
+    /// Returns error if:
+    /// - The deposit amount doesn't match the bridge's expected amount
+    /// - The DRT spending signature is invalid or doesn't match the aggregated operator key
+    /// - The deposit output lock is incorrect
+    /// - A deposit with the same index already exists
+    fn validate_deposit(
+        &self,
+        tx: &Transaction,
+        info: &DepositInfo,
+    ) -> Result<(), DepositValidationError> {
+        // Verify the deposit amount matches the bridge's expected amount
+        if info.amt.to_sat() != self.denomination.to_sat() {
+            return Err(DepositValidationError::MismatchDepositAmount(Mismatch {
+                expected: self.denomination.to_sat(),
+                got: info.amt.to_sat(),
+            }));
+        }
+
+        // Validate the DRT spending signature against the aggregated operator key
+        validate_drt_spending_signature(
+            tx,
+            info.drt_tapscript_merkle_root,
+            self.operators().agg_key(),
+            info.amt.into(),
+        )?;
+
+        // Ensure the deposit output is properly locked to the aggregated operator key
+        validate_deposit_output_lock(tx, self.operators().agg_key())?;
+
+        // Verify this deposit index hasn't been used before
+        if self.deposits().get_deposit(info.deposit_idx).is_some() {
+            return Err(DepositValidationError::DepositIdxAlreadyExists(
+                info.deposit_idx,
+            ));
+        }
+
+        Ok(())
+    }
+
+    /// Creates a withdrawal assignment by selecting an unassigned deposit and assigning it to an
+    /// operator.
+    ///
+    /// This function handles incoming withdrawal commands by:
+    /// 1. Finding a deposit that has not been assigned yet
+    /// 2. Randomly selecting an operator from the current multisig set that is also a notary
+    ///    operator for that deposit
+    /// 3. Creating an assignment linking the deposit to the selected operator with a deadline
+    ///    calculated from the current block height plus the configured deadline duration
+    ///
+    /// # Parameters
+    ///
+    /// - `withdrawal_cmd` - The withdrawal command specifying outputs and amounts
+    /// - `l1_block_id` - The L1 block ID used as seed for random operator selection
+    /// - `current_block_height` - The current Bitcoin block height for deadline calculation
```

Comment:
why are these different?

Reply:
The question was why we need two argument for the bitcoin block, which I believe you now consolidated.